docs/nebula-design-concepts.md

```diff
@@ -0,0 +1,213 @@
+
+In the preceding example, there are two kinds of edges, one is **like** pointing
+from player to player, the property is likeness (double); the other is **serve**,
+pointing from player to team, the properties are start_year(int) and end_year(int).
+
+It should be noted that there may be **multiple edges of the same or different types
+at the same time** between any two vertices.
+
+
+
+## Graph partition
+
+![image](https://user-images.githubusercontent.com/42762957/61119934-76829b80-a4ce-11e9-9d49-2abb11b5f7b2.png)
+
+
+In applications such as social networks, or business traction networks, the number of vertices and their connections are extremely large. Therefore, partitioning large graphs is inevitable for complexity reduction and parallelization. **Nebula Graph** adopts **Edge Separators** to cut a graph into smaller components. And the partition number cannot be changed once being configured.
+
+## Data model
+
+
+![image](https://user-images.githubusercontent.com/42762957/61120073-b0ec3880-a4ce-11e9-975f-c19482d4b109.png)
+
+
+
+Data is stored as key-value pairs in **Nebula Graph**. For vertices, as illustrated in the diagram above, the key consists of four parts: partID (4 bytes), vertexID (8 bytes), tagID (4 bytes) and Version (8 bytes). And they are spread across various partitions based on the hash of their vertexIds (vid).
+
+
+Each edge in **Nebula Graph** is stored twice. One is stored in the same partition as the source vertex, namely out-edge. The other one is stored in the same partition as the destination vertex, namely in-edge. Only the out-edge carries the properties.
+
+The key format for the out-edge is: partID (4 bytes), sourceVertexID (8 bytes), edgeType (4 bytes), ranking (8 bytes), destinationVertexID (8 bytes)
+
+The key format for the in-edge is: partID (4 bytes), destinationVertexID (8 bytes), -edgeType (4 bytes), ranking (8 bytes), sourceVertexID (8 bytes)
+
+
+![image](https://user-images.githubusercontent.com/42762957/61120260-1c360a80-a4cf-11e9-8a43-8c4ca2d73572.png)
+
+
+
+## Architecture
+
+**Nebula Graph** includes four modules: storage service, meta service, query engine and client.
+
+
+![image](https://user-images.githubusercontent.com/42762957/61120288-31ab3480-a4cf-11e9-9905-a1d4b1e6c523.png)
+
+
+
+
+### Storage service
+
+The corresponding process of storage service is nebula-storaged, which provides a key-value store. Separated storage and computing makes storage service flexible, and multiple storage engines like RocksDB and HBase are supported, with RocksDB set as default engine. To build a resilient distributed system, [Raft](https://raft.github.io/) is implemented as the consensus algorithm. Currently **Nebula Graph** supports
+multiple storage engines like Rocksdb and HBase, etc.
```

Comment:
Currently Nebula Graph supports multiple storage engines like Rocksdb and HBase, etc.   -- you can delete. duplicate